src/main/java/com/drajer/bsa/kar/action/SubmitReport.java

```diff
@@ -51,47 +56,51 @@ public BsaActionStatus process(KarProcessingData data, EhrQueryService ehrServic
+      try (InputStream inputStream =
+          SubmitReport.class.getClassLoader().getResourceAsStream("report-headers.properties")) {
+        Properties headers = new Properties();
+        headers.load(inputStream);
+        for (Resource r : resourcesToSubmit) {
 
-        IGenericClient client = context.newRestfulGenericClient(submissionEndpoint);
+          IGenericClient client = context.newRestfulGenericClient(submissionEndpoint);
 
-        context.getRestfulClientFactory().setSocketTimeout(30 * 1000);
+          context.getRestfulClientFactory().setSocketTimeout(30 * 1000);
 
-        // All submissions are expected to be bundles
-        Bundle bundleToSubmit = (Bundle) r;
+          // All submissions are expected to be bundles
+          Bundle bundleToSubmit = (Bundle) r;
+          IOperationProcessMsgMode<IBaseResource> operation =
+              client.operation().processMessage().setMessageBundle(bundleToSubmit);
+          headers.forEach(
+              (key, value) -> operation.withAdditionalHeader((String) key, (String) value));
+          Bundle responseBundle = (Bundle) operation.encodedJson().execute();
 
-        Bundle responseBundle =
-            (Bundle)
-                client
-                    .operation()
-                    .processMessage()
-                    .setMessageBundle(bundleToSubmit)
-                    .encodedJson()
-                    .execute();
+          if (responseBundle != null) {
+            logger.info(
+                "Response Bundle:::::{}",
+                context.newJsonParser().encodeResourceToString(responseBundle));
 
-        if (responseBundle != null) {
-          logger.info(
-              "Response Bundle:::::{}",
-              context.newJsonParser().encodeResourceToString(responseBundle));
+            data.addActionOutput(actionId, responseBundle);
 
-          data.addActionOutput(actionId, responseBundle);
+            logger.info(" Adding Response Bundle to output using id {}", responseBundle.getId());
 
-          logger.info(" Adding Response Bundle to output using id {}", responseBundle.getId());
+            data.addActionOutputById(responseBundle.getId(), responseBundle);
+          }
 
-          data.addActionOutputById(responseBundle.getId(), responseBundle);
-        }
+          if (conditionsMet(data)) {
 
-        if (conditionsMet(data)) {
+            // Execute sub Actions
+            executeSubActions(data, ehrService);
 
-          // Execute sub Actions
-          executeSubActions(data, ehrService);
+            // Execute Related Actions.
+            executeRelatedActions(data, ehrService);
+          }
 
-          // Execute Related Actions.
-          executeRelatedActions(data, ehrService);
-        }
+          actStatus.setActionStatus(BsaActionStatusType.Completed);
+        } // for all resources to be submitted
+      } catch (IOException ex) {
+        logger.error("Error while loading Action Classes from Proporties File ");
```

Comment:
Can this message include the actual exception? And I think this was copy&pasted so it should refer to report headers not action classes.